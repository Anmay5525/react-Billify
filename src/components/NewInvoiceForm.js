import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import InvoicesFormItem from "./InvoicesFormItem";
import Loader from "./Loader";
import DropDown from "./DropDown";

export default function NewInvoiceForm({ handleSubRouteChange }) {
  const [isVisibleItem, setIsVisibleItem] = useState(false);
  const [isVisibleCustomer, setisVisibleCustomer] = useState(true);
  const [dummy, setDummy] = useState(false);
  const [availableItems, setAvailableItems] = useState(null);
  const [availableCustomers, setAvailableCustomers] = useState(null);
  const [selectedCustomer, setselectedCustomer] = useState(null);
  const [selectedItems, setSelectedItems] = useState([
    // { name: "model S", price: 100, id: "sasfwe", quantity: 1, amount: 100 },
    // {
    //   name: "modelasdasdasd S",
    //   price: 1003,
    //   id: "hgjhg",
    //   quantity: 1,
    //   amount: 1003,
    // },
  ]);
  const handleNewInvoice = (e) => {
    e.preventDefault();

    if (!selectedCustomer) {
      toast.warn("Please select a customer!!");
    }
    if (!selectedItems.length) {
      toast.warn("Please select atleast 1 item!!");
    }

    if (selectedItems.length && selectedCustomer) {
      const dueDate = new Date(e.target.due.value).getTime();

      const url = "https://rzp-training.herokuapp.com/team1/invoices";

      const inv = {
        customer_id: selectedCustomer.id,
        line_items: [],

        // send date in seconds
        expire_by: Math.round(dueDate / 1000),
      };
      selectedItems.forEach((item) => {
        inv.line_items.push({
          item_id: item.id,
          quantity: item.quantity,
        });
      });

      //   console.log(inv);

      fetch(url, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(inv),
      })
        .then((res) => {
          if (res.ok) {
            return res.json();
          }
          return toast(
            `Something went wrong. Server responded with status ${res.status}`
          );
        })
        .then((r) => {
          if (r) {
            toast.success("New invoice created");
            handleSubRouteChange("list");
          }
        })
        .catch((error) => console.log(error));
    }
  };

  const handleItemSelect = (e) => {
    const selected = JSON.parse(e.target.value);
    // if already selected an item then do nothing
    if (
      !selectedItems.find((el) => {
        return el.id === selected.id;
      })
    ) {
      const obj = [
        ...selectedItems,
        {
          name: selected.name,
          price: selected.unit_amount / 100,
          id: selected.id,
          quantity: 1,
          amount: selected.unit_amount / 100,
        },
      ];
      setSelectedItems(obj);
      setIsVisibleItem(false);
    } else {
      toast.warn("Item already added. Please select some other item!");
    }
  };

  const handleCustomerSelect = (e) => {
    const selected = JSON.parse(e.target.value);

    setselectedCustomer(selected);
    setisVisibleCustomer(false);
  };

  const handleQuantityChange = (quant, index) => {
    const cp = { ...selectedItems[index] };
    cp.quantity = Number(quant);
    cp.amount = Number((cp.quantity * cp.price).toFixed(2));

    setSelectedItems((prev) => {
      // eslint-disable-next-line no-param-reassign
      prev[index] = cp;
      return prev;
    });

    setDummy((prev) => !prev);
  };

  const handleDelete = (index) => {
    const cp = [...selectedItems];
    cp.splice(index, 1);
    setSelectedItems(cp);
  };

  const controller = new AbortController();
  const { signal } = controller;

  useEffect(() => {
    fetch("https://rzp-training.herokuapp.com/team1/items", { signal })
      .then((res) => res.json())
      .then((r) => setAvailableItems(r))
      .catch((er) => console.log(er));

    fetch("https://rzp-training.herokuapp.com/team1/customers", { signal })
      .then((res) => res.json())
      .then((r) => setAvailableCustomers(r))
      .catch((er) => console.log(er));

    return () => {
      controller.abort();
    };
  }, []);
  if (availableCustomers && availableItems) {
    let total = 0;
    selectedItems.forEach((el) => {
      total += el.amount;
      total = Number(total.toFixed(2));
    });

    // console.log(selectedItems, total);
    return (
      <div className="content">
        {/* <ToastContainer /> */}
        <form onSubmit={(e) => handleNewInvoice(e)}>
          <div className="invoices-title-container">
            <div className="invoices-title">Invoices</div>
            <button type="submit" className="invoices-new-btn">
              Save Invoice
            </button>
          </div>
          <div className="new-invoice-customer-details">
            <div className="billto-cont">
              <span>Bill to</span>
              <div className="billto-details">
                <div
                  style={{
                    display: "flex",
                    flexDirection: "column",
                    border: "none",
                    resize: "none",
                    padding: "4px 4px 4px 4px",
                    fontSize: "0.9em",
                  }}
                >
                  <span>{selectedCustomer ? selectedCustomer.name : ""}</span>
                  <span>
                    {selectedCustomer ? selectedCustomer.contact : ""}
                  </span>
                  <span>{selectedCustomer ? selectedCustomer.email : ""}</span>
                </div>
                {!isVisibleCustomer ? (
                  <button
                    type="button"
                    className="billto-change-btn"
                    onClick={() => setisVisibleCustomer(true)}
                  >
                    Change
                  </button>
                ) : (
                  ""
                )}
              </div>
              {isVisibleCustomer ? (
                <DropDown
                  handleSelect={handleCustomerSelect}
                  list={availableCustomers.items}
                />
              ) : (
                ""
              )}
            </div>

            <div style={{ display: "flex", flexDirection: "column" }}>
              <span>Issued At</span>
              <input type="date" required name="issue" className="date" />
            </div>
            <div style={{ display: "flex", flexDirection: "column" }}>
              <span>Due Date</span>
              <input type="date" required name="due" className="date" />
            </div>
          </div>
          <div className="new-invoice-table-cont">
            <table className="new-invoice-table">
              <thead>
                <tr>
                  <th>Items</th>
                  <th>Quantity</th>
                  <th>Price</th>
                  <th>Amount</th>
                </tr>
              </thead>
              <tbody>
                {selectedItems.map((item, i) => {
                  return (
                    <InvoicesFormItem
                      index={i}
                      data={item}
                      handleQuantityChange={handleQuantityChange}
                      handleDelete={handleDelete}
                      key={item.name + window.performance.now()}
                    />
                  );
                })}
              </tbody>
            </table>
            <button
              type="button"
              className="simple-btn"
              onClick={() => setIsVisibleItem(true)}
            >
              Add Item
            </button>
            <div className="select-cnt">
              {isVisibleItem ? (
                <DropDown
                  handleSelect={handleItemSelect}
                  list={availableItems.items}
                />
              ) : (
                ""
              )}
            </div>
          </div>
        </form>
        <div className="invoices-info-container">
          <div style={{ display: "flex", flexDirection: "column" }}>
            <span>Notes</span>
            <textarea
              style={{ width: "80%" }}
              className="new-invoice-textarea"
              name="notes"
              rows="4"
            />
          </div>
          <div className="invoices-info">
            <table className="new-invoice-info-table">
              <tbody>
                {selectedItems.map((item) => {
                  return (
                    <tr key={item.amount + window.performance.now()}>
                      <td>{item.name}</td>
                      <td align="right">
                        <span>x</span>
                        {item.quantity}
                      </td>
                      <td align="right">
                        <span>&#8377;</span>
                        {item.amount}
                      </td>
                    </tr>
                  );
                })}
              </tbody>
            </table>

            <hr width="100%" />
            <div
              style={{
                padding: "0px 10px 0px 6px",
                fontSize: "0.92em",
                display: "flex",
                justifyContent: "space-between",
              }}
            >
              <span>Total amount:</span>
              <span style={{ color: "#5851d7" }}>
                <span>&#8377;</span>
                {total}
              </span>
            </div>
          </div>
        </div>
      </div>
    );
  }
  return <Loader />;
}

NewInvoiceForm.propTypes = {
  handleSubRouteChange: PropTypes.func.isRequired,
};
