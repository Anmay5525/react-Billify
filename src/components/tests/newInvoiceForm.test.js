import React from 'react';
import { configure, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import NewInvoiceForm from '../NewInvoiceForm';
import Loader from '../Loader';

configure({adapter:new Adapter()});


describe('<NewInvoiceForm/>',()=>{
    it('renders loader image when in data is not loaded',()=>{
        const tree=shallow(<NewInvoiceForm />);
        expect(tree.find(Loader)).toHaveLength(1);
    });

    it("",()=>{
        
    });
});