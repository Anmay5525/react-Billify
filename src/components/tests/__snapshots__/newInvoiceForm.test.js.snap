// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewInvoiceForm/> handles customer select properly 1`] = `"<div class=\\"content\\"><form><div class=\\"invoices-title-container\\"><div class=\\"invoices-title\\">New Invoice</div><button type=\\"submit\\" class=\\"invoices-new-btn\\">Save Invoice</button></div><div class=\\"new-invoice-customer-details\\"><div class=\\"billto-cont\\"><span>Bill to</span><div class=\\"billto-details\\"><div style=\\"display: flex; flex-direction: column; resize: none; padding: 4px 4px 4px 4px; font-size: 0.9em;\\"><span>John Stones</span><span>2837495867</span><span>s@s.com</span></div><button type=\\"button\\" class=\\"billto-change-btn\\">Change</button></div></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Issued At</span><input type=\\"date\\" required=\\"\\" name=\\"issue\\" class=\\"date\\"></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Due Date</span><input type=\\"date\\" required=\\"\\" name=\\"due\\" class=\\"date\\"></div></div><div class=\\"new-invoice-table-cont\\"><table class=\\"new-invoice-table\\"><thead><tr><th>Items</th><th>Quantity</th><th>Price</th><th>Amount</th></tr></thead><tbody></tbody></table><button type=\\"button\\" class=\\"simple-btn add-item-btn\\">Add Item</button><div class=\\"select-cnt\\"></div></div></form><div class=\\"invoices-info-container\\"><div style=\\"display: flex; flex-direction: column;\\"><span>Notes</span><textarea style=\\"width: 80%;\\" class=\\"new-invoice-textarea\\" name=\\"notes\\" rows=\\"4\\"></textarea></div><div class=\\"invoices-info\\"><table class=\\"new-invoice-info-table\\"><tbody></tbody></table><hr width=\\"100%\\"><div style=\\"padding: 0px 10px 0px 6px; font-size: 0.92em; display: flex; justify-content: space-between;\\"><span>Total amount:</span><span style=\\"color: rgb(88, 81, 215);\\"><span>₹</span>0</span></div></div></div></div>"`;

exports[`<NewInvoiceForm/> handles item functionality correctly 1`] = `"<div class=\\"content\\"><form><div class=\\"invoices-title-container\\"><div class=\\"invoices-title\\">New Invoice</div><button type=\\"submit\\" class=\\"invoices-new-btn\\">Save Invoice</button></div><div class=\\"new-invoice-customer-details\\"><div class=\\"billto-cont\\"><span>Bill to</span><div class=\\"billto-details\\"><div style=\\"display: flex; flex-direction: column; resize: none; padding: 4px 4px 4px 4px; font-size: 0.9em;\\"><span></span><span></span><span></span></div></div><select style=\\"width: 100%; height: 26px;\\"><option class=\\"option\\">Select Item</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_Fg0tl7Y2qra6SJ&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;David Silva&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2938475867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600764149}\\">David Silva</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_FfeM7EaKLd8SOh&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;John Stones&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2837495867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600684763}\\">John Stones</option></select></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Issued At</span><input type=\\"date\\" required=\\"\\" name=\\"issue\\" class=\\"date\\"></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Due Date</span><input type=\\"date\\" required=\\"\\" name=\\"due\\" class=\\"date\\"></div></div><div class=\\"new-invoice-table-cont\\"><table class=\\"new-invoice-table\\"><thead><tr><th>Items</th><th>Quantity</th><th>Price</th><th>Amount</th></tr></thead><tbody><tr class=\\"tr\\"><td>Kora Kagaz</td><td align=\\"center\\"><input class=\\"new-invoice-quantity\\" type=\\"number\\" name=\\"quantity\\" min=\\"1\\" max=\\"5\\" required=\\"\\" value=\\"1\\"></td><td><span>₹</span>10</td><td><span>₹</span>10</td><td><button style=\\"font-size: 0.8em;\\" type=\\"button\\" class=\\"simple-btn\\">Delete</button></td></tr></tbody></table><button type=\\"button\\" class=\\"simple-btn add-item-btn\\">Add Item</button><div class=\\"select-cnt\\"></div></div></form><div class=\\"invoices-info-container\\"><div style=\\"display: flex; flex-direction: column;\\"><span>Notes</span><textarea style=\\"width: 80%;\\" class=\\"new-invoice-textarea\\" name=\\"notes\\" rows=\\"4\\"></textarea></div><div class=\\"invoices-info\\"><table class=\\"new-invoice-info-table\\"><tbody><tr><td>Kora Kagaz</td><td align=\\"right\\"><span>x</span>1</td><td align=\\"right\\"><span>₹</span>10</td></tr></tbody></table><hr width=\\"100%\\"><div style=\\"padding: 0px 10px 0px 6px; font-size: 0.92em; display: flex; justify-content: space-between;\\"><span>Total amount:</span><span style=\\"color: rgb(88, 81, 215);\\"><span>₹</span>10</span></div></div></div></div>"`;

exports[`<NewInvoiceForm/> handles item functionality correctly 2`] = `"<div class=\\"content\\"><form><div class=\\"invoices-title-container\\"><div class=\\"invoices-title\\">New Invoice</div><button type=\\"submit\\" class=\\"invoices-new-btn\\">Save Invoice</button></div><div class=\\"new-invoice-customer-details\\"><div class=\\"billto-cont\\"><span>Bill to</span><div class=\\"billto-details\\"><div style=\\"display: flex; flex-direction: column; resize: none; padding: 4px 4px 4px 4px; font-size: 0.9em;\\"><span></span><span></span><span></span></div></div><select style=\\"width: 100%; height: 26px;\\"><option class=\\"option\\">Select Item</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_Fg0tl7Y2qra6SJ&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;David Silva&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2938475867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600764149}\\">David Silva</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_FfeM7EaKLd8SOh&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;John Stones&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2837495867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600684763}\\">John Stones</option></select></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Issued At</span><input type=\\"date\\" required=\\"\\" name=\\"issue\\" class=\\"date\\"></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Due Date</span><input type=\\"date\\" required=\\"\\" name=\\"due\\" class=\\"date\\"></div></div><div class=\\"new-invoice-table-cont\\"><table class=\\"new-invoice-table\\"><thead><tr><th>Items</th><th>Quantity</th><th>Price</th><th>Amount</th></tr></thead><tbody><tr class=\\"tr\\"><td>Kora Kagaz</td><td align=\\"center\\"><input class=\\"new-invoice-quantity\\" type=\\"number\\" name=\\"quantity\\" min=\\"1\\" max=\\"5\\" required=\\"\\" value=\\"1\\"></td><td><span>₹</span>10</td><td><span>₹</span>10</td><td><button style=\\"font-size: 0.8em;\\" type=\\"button\\" class=\\"simple-btn\\">Delete</button></td></tr><tr class=\\"tr\\"><td>disc</td><td align=\\"center\\"><input class=\\"new-invoice-quantity\\" type=\\"number\\" name=\\"quantity\\" min=\\"1\\" max=\\"5\\" required=\\"\\" value=\\"1\\"></td><td><span>₹</span>300</td><td><span>₹</span>300</td><td><button style=\\"font-size: 0.8em;\\" type=\\"button\\" class=\\"simple-btn\\">Delete</button></td></tr></tbody></table><button type=\\"button\\" class=\\"simple-btn add-item-btn\\">Add Item</button><div class=\\"select-cnt\\"></div></div></form><div class=\\"invoices-info-container\\"><div style=\\"display: flex; flex-direction: column;\\"><span>Notes</span><textarea style=\\"width: 80%;\\" class=\\"new-invoice-textarea\\" name=\\"notes\\" rows=\\"4\\"></textarea></div><div class=\\"invoices-info\\"><table class=\\"new-invoice-info-table\\"><tbody><tr><td>Kora Kagaz</td><td align=\\"right\\"><span>x</span>1</td><td align=\\"right\\"><span>₹</span>10</td></tr><tr><td>disc</td><td align=\\"right\\"><span>x</span>1</td><td align=\\"right\\"><span>₹</span>300</td></tr></tbody></table><hr width=\\"100%\\"><div style=\\"padding: 0px 10px 0px 6px; font-size: 0.92em; display: flex; justify-content: space-between;\\"><span>Total amount:</span><span style=\\"color: rgb(88, 81, 215);\\"><span>₹</span>310</span></div></div></div></div>"`;

exports[`<NewInvoiceForm/> handles item functionality correctly 3`] = `"<div class=\\"content\\"><form><div class=\\"invoices-title-container\\"><div class=\\"invoices-title\\">New Invoice</div><button type=\\"submit\\" class=\\"invoices-new-btn\\">Save Invoice</button></div><div class=\\"new-invoice-customer-details\\"><div class=\\"billto-cont\\"><span>Bill to</span><div class=\\"billto-details\\"><div style=\\"display: flex; flex-direction: column; resize: none; padding: 4px 4px 4px 4px; font-size: 0.9em;\\"><span></span><span></span><span></span></div></div><select style=\\"width: 100%; height: 26px;\\"><option class=\\"option\\">Select Item</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_Fg0tl7Y2qra6SJ&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;David Silva&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2938475867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600764149}\\">David Silva</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_FfeM7EaKLd8SOh&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;John Stones&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2837495867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600684763}\\">John Stones</option></select></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Issued At</span><input type=\\"date\\" required=\\"\\" name=\\"issue\\" class=\\"date\\"></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Due Date</span><input type=\\"date\\" required=\\"\\" name=\\"due\\" class=\\"date\\"></div></div><div class=\\"new-invoice-table-cont\\"><table class=\\"new-invoice-table\\"><thead><tr><th>Items</th><th>Quantity</th><th>Price</th><th>Amount</th></tr></thead><tbody><tr class=\\"tr\\"><td>disc</td><td align=\\"center\\"><input class=\\"new-invoice-quantity\\" type=\\"number\\" name=\\"quantity\\" min=\\"1\\" max=\\"5\\" required=\\"\\" value=\\"1\\"></td><td><span>₹</span>300</td><td><span>₹</span>300</td><td><button style=\\"font-size: 0.8em;\\" type=\\"button\\" class=\\"simple-btn\\">Delete</button></td></tr></tbody></table><button type=\\"button\\" class=\\"simple-btn add-item-btn\\">Add Item</button><div class=\\"select-cnt\\"></div></div></form><div class=\\"invoices-info-container\\"><div style=\\"display: flex; flex-direction: column;\\"><span>Notes</span><textarea style=\\"width: 80%;\\" class=\\"new-invoice-textarea\\" name=\\"notes\\" rows=\\"4\\"></textarea></div><div class=\\"invoices-info\\"><table class=\\"new-invoice-info-table\\"><tbody><tr><td>disc</td><td align=\\"right\\"><span>x</span>1</td><td align=\\"right\\"><span>₹</span>300</td></tr></tbody></table><hr width=\\"100%\\"><div style=\\"padding: 0px 10px 0px 6px; font-size: 0.92em; display: flex; justify-content: space-between;\\"><span>Total amount:</span><span style=\\"color: rgb(88, 81, 215);\\"><span>₹</span>300</span></div></div></div></div>"`;

exports[`<NewInvoiceForm/> renders loader image when in data is loaded 1`] = `"<div class=\\"content\\"><form><div class=\\"invoices-title-container\\"><div class=\\"invoices-title\\">New Invoice</div><button type=\\"submit\\" class=\\"invoices-new-btn\\">Save Invoice</button></div><div class=\\"new-invoice-customer-details\\"><div class=\\"billto-cont\\"><span>Bill to</span><div class=\\"billto-details\\"><div style=\\"display: flex; flex-direction: column; resize: none; padding: 4px 4px 4px 4px; font-size: 0.9em;\\"><span></span><span></span><span></span></div></div><select style=\\"width: 100%; height: 26px;\\"><option class=\\"option\\">Select Item</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_Fg0tl7Y2qra6SJ&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;David Silva&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2938475867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600764149}\\">David Silva</option><option class=\\"option\\" value=\\"{&quot;id&quot;:&quot;cust_FfeM7EaKLd8SOh&quot;,&quot;entity&quot;:&quot;customer&quot;,&quot;name&quot;:&quot;John Stones&quot;,&quot;email&quot;:&quot;s@s.com&quot;,&quot;contact&quot;:&quot;2837495867&quot;,&quot;gstin&quot;:null,&quot;notes&quot;:[],&quot;created_at&quot;:1600684763}\\">John Stones</option></select></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Issued At</span><input type=\\"date\\" required=\\"\\" name=\\"issue\\" class=\\"date\\"></div><div style=\\"display: flex; flex-direction: column; width: 26%;\\"><span>Due Date</span><input type=\\"date\\" required=\\"\\" name=\\"due\\" class=\\"date\\"></div></div><div class=\\"new-invoice-table-cont\\"><table class=\\"new-invoice-table\\"><thead><tr><th>Items</th><th>Quantity</th><th>Price</th><th>Amount</th></tr></thead><tbody></tbody></table><button type=\\"button\\" class=\\"simple-btn add-item-btn\\">Add Item</button><div class=\\"select-cnt\\"></div></div></form><div class=\\"invoices-info-container\\"><div style=\\"display: flex; flex-direction: column;\\"><span>Notes</span><textarea style=\\"width: 80%;\\" class=\\"new-invoice-textarea\\" name=\\"notes\\" rows=\\"4\\"></textarea></div><div class=\\"invoices-info\\"><table class=\\"new-invoice-info-table\\"><tbody></tbody></table><hr width=\\"100%\\"><div style=\\"padding: 0px 10px 0px 6px; font-size: 0.92em; display: flex; justify-content: space-between;\\"><span>Total amount:</span><span style=\\"color: rgb(88, 81, 215);\\"><span>₹</span>0</span></div></div></div></div>"`;

exports[`<NewInvoiceForm/> renders loader image when in data is not loaded 1`] = `"<div class=\\"loader-container\\"><svg viewBox=\\"0 0 100 100\\" xmlns=\\"http://www.w3.org/2000/svg\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"45\\"></circle></svg></div>"`;
